### ISSUE TITLE
Lit Integration Boost

### ISSUE THREAD

**Issue by skuhlmann** [odyssy-automaton/pokemol-web#1774]
_**Need some details here**_

Work with the lit team on an integration: daohaus <> lit. Likely dao permissioned google docs.

[existing pr](https://github.com/HausDAO/daohaus-app/compare/feature/lit-boost?expand=1)

this boost can be modeled similar to the snapshot boost - potentially showing a link to `Drive Links` on the main Dao page which would link to the LitBoost page where all the google drive doc links can be displayed.

Integration scope: 
* On boosts page, a "Share Google Drive Item" button that takes them to our Google Drive dashboard, with some stuff prefilled from URL variables
* On boosts page, a list of the Google Drive files that were shared by anyone in the DAO

Data to send to Lit as URL variables when the user clicks on the "Share Google Drive Item" button:
* Contract Address
* Contract Type (molochdao v2.1)
* DAO Name

Functionality of boost:
1.  Boost is added to dao
2. User can sign up to google lit oauth
3. Which unlocks a table of shareable links for the entire dao to access 

fetching shareable links (Lit Docs):

1) simply check if the user is a member of a dao: https://developer.litprotocol.com/docs/AccessControlConditions/evmBasicExamples#must-be-a-member-of-a-dao-molochdaov21-also-supports-daohaus
2) more advanced options where you can check the number of shares a user has in a DAO or anything the members() function returns: https://developer.litprotocol.com/docs/AccessControlConditions/evmCustomContractCalls#must-have-at-least-1-share-in-a-given-molochdaov21-dao

relavent resources:

https://github.com/LIT-Protocol/lit-oauth/tree/main/server - backend service of the Lit protocol - navigate to google for relavent backend service for this story.

https://gist.github.com/glitch003/76ac07a95cd11ddc1571254f59682665 - gist for interacting with Lit
https://github.com/LIT-Protocol/lit-oauth/blob/main/server/oauth/google.js#L398-L407 -  To get all shares from a DAO
https://github.com/LIT-Protocol/lit-oauth/blob/main/server/oauth/google.js#L225-L237 -  create a share of a google drive file

supported `LIT_CHAINS` - https://github.com/LIT-Protocol/lit-js-sdk/blob/main/src/lib/constants.js#L30

**Comment by earth2travis** [comment]
Waiting on a response from Lit team

**Comment by earth2travis** [comment]
Needs to have some APIs added to their service

**Comment by earth2travis** [comment]
Add OAuth to API

**Comment by alexkeating** [comment]
They expect to have the oauth changes ready by the end of the week

**Comment by earth2travis** [comment]
Necessary endpoints have been added. Should be good to go

**Comment by brossetti1** [comment]
updated the description of this one with more information - should be able to get to work on it tomorrow

**Comment by brossetti1** [comment]
currently, I see this as the implementation plan for this integration:

we can redirect to the auth through the Lit UI and on our end, use the authSig to get the user profile and token (if needed - maybe for unsharing files from daohaus).

1. get auth sig after setting up boost:
```
currentAuthSig = await LitJsSdk.checkAndSignAuthMessage({ chain });

// supported `LIT_CHAINS` - https://github.com/LIT-Protocol/lit-js-sdk/blob/main/src/lib/constants.js#L30
// store on our side - most likely local storage
```
2. check user is authenticated and fetch shared files

```
checkIfUserExists = async (authSig) => {
  // https://github.com/LIT-Protocol/lit-oauth/blob/51b6efc4c45ee6b0bf0ebfed4f8713c6c045b954/server/oauth/google.js#L172-L194
  return await axios.post(`${LIT_API_HOST}/api/google/checkIfUserExists`, {
    authSig,
  });
};

const userExists = await checkIfUserExists(authSig);
console.log("check frontend userExists", userExists);

if (!userExists.data) {
  // if no lit user exists, redirect to authenticate
  //  redirect to Lit UI for auth
} else {
  await handleLoadCurrentUser(authSig);
  // OR await getAllShares(authSig, userProfile.idOnService);
}

// if accessToken is necessary for certain calls (potentially "unshare" document)
const handleLoadCurrentUser = async (authSig) => {
  const userInfo = await getUserProfile(authSig);
  // check for google drive scope and sign user out if scope is not present
  if (
    userInfo.data["scope"] &&
    userInfo.data["scope"].includes(
      "https://www.googleapis.com/auth/drive.file"
    )
  ) {
    const profileData = JSON.parse(userInfo.data.extraData);
    const accessToken = profileData.accessToken; // if we need jwt access token for some api call
    const userProfile = {
      idOnService: userInfo.data.idOnService,
      email: userInfo.data.email,
      displayName: profileData.displayName,
      avatar: profileData.photoLink,
    };

    await getAllShares(authSig, userProfile.idOnService); <--- shared files loaded here
  }
}
```

the user will have to share the documents and link it to the dao address **manually** as far as I can see - the oauth through the UI and especially the Lit Access Control modal that is used in the UI does not currently fill in the dao address based on url params or else I would expect default props to be set as the `DAOAddress`

v2 implementation - https://github.com/LIT-Protocol/lit-share-modal-v2/blob/c17c0804246c61c58db591e8729cf51c43d4c5d5/src/conditionCreationFlow/SelectDAO.jsx

v3 implementation - https://github.com/LIT-Protocol/lit-share-modal-v3/blob/d2f81d7b608d20cc4573353872c1530d92156214/src/chainComponents/ethereum/ethereumConditionCreationFlow/EthereumSelectDAO.jsx

this functionality should be implemented by the lit team so we can pass the correct params for sharing docs with the specific dao as specified here:

```
Data to send to Lit as URL variables when the user clicks on the "Share Google Drive Item" button:

Contract Address
Contract Type (molochdao v2.1)
DAO Name
```

**Comment by earth2travis** [comment]
Need to determine if their servers are not allowing signatures from other chains

**Comment by brossetti1** [comment]
not sure why the pull request is not linked:

https://github.com/HausDAO/daohaus-app/pull/1897

Just an update here - last communication we worked out some bugs on Lit Protocols end with `checkifUserExists` - i have added a feature flag (`REACT_APP_DEV`) check so this does not load on staging or production. We are going to get this in for now and revisit when Lit gets back to us

**Comment by plor** [comment]
If you want to auto-link an issue, the PR needs to say "fixes #1774" otherwise it is just in the stream. I've been manually linking when I can.
